Jint/Runtime/Descriptors/Specialized/IndexDescriptor.cs

```diff
@@ -97,12 +97,19 @@ protected internal override JsValue CustomValue
                 {
                     return JsValue.FromObject(_engine, getter.Invoke(_target, parameters));
                 }
-                catch
+                catch (TargetInvocationException tie)
                 {
-                    return JsValue.Undefined;
+                    switch (tie.InnerException)
+                    {
+                        case null:
+                            throw;
+                        case ArgumentOutOfRangeException _:

```

Comment:
Why is this one special? Can there be other exceptions that would need to behave the same?

Reply:
I think the original idea with the exception swallowing was to behave like JavaScript, over or under-indexing should produce undefined. There could be more exceptions to "swallow", but hard to say.

Reply:
I added `IndexOutOfRangeException` as that is what happens with native array, List<T> throws `ArgumentOutOfRangeException`, good catch